test/integration/tests/backstop.ts

```diff
@@ -0,0 +1,93 @@
+const toBN = ethers.BigNumber.from;
+
+before(async () => {
+  chai.use(CBN(ethers.BigNumber));
+  chai.use(solidity);
+  await resetFork();
+});
+
+describe.only('e2e', function () {
+  let contracts: NamedContracts;
+  let contractAddresses: NamedAddresses;
+  let deployAddress: string;
+  let e2eCoord: TestEndtoEndCoordinator;
+  let doLogging: boolean;
+
+  const tenPow18 = toBN('1000000000000000000');
+
+  before(async function () {
+    // Setup test environment and get contracts
+    const version = 1;
+    deployAddress = (await ethers.getSigners())[0].address;
+    if (!deployAddress) throw new Error(`No deploy address!`);
+
+    doLogging = Boolean(process.env.LOGGING);
+
+    const config = {
+      logging: doLogging,
+      deployAddress: deployAddress,
+      version: version
+    };
+
+    e2eCoord = new TestEndtoEndCoordinator(config, proposals);
+
+    doLogging && console.log(`Loading environment...`);
+    ({ contracts, contractAddresses } = await e2eCoord.loadEnvironment());
+    doLogging && console.log(`Environment loaded.`);
+  });
+
+  describe('TribeMinter', async function () {
+    it('mint TRIBE', async function () {
+      const { tribeMinter, tribe } = contracts;
+      const tribeSupply = await tribe.totalSupply();
+      const balanceBefore = await tribe.balanceOf(deployAddress);
+
+      await tribeMinter.mint(deployAddress, '100000');
+
+      // Minting increases total supply and target balance
+      expect(balanceBefore.add(toBN('100000'))).to.be.equal(await tribe.balanceOf(deployAddress));
+      expect(tribeSupply.add(toBN('100000'))).to.be.equal(await tribe.totalSupply());
+    });
+  });
+  describe('TribeReserveStabilizer', async function () {
+    it('exchangeFei', async function () {
+      const { fei, staticPcvDepositWrapper, tribe, tribeReserveStabilizer, collateralizationOracleWrapper } = contracts;
+
+      await fei.mint(deployAddress, tenPow18.mul(tenPow18).mul(toBN(4)));
+      await collateralizationOracleWrapper.update();
+
+      const userFeiBalanceBefore = toBN(await fei.balanceOf(deployAddress));
+      const userTribeBalanceBefore = await tribe.balanceOf(deployAddress);
+
+      const feiTokensExchange = toBN(40000000000000);
+      await fei.approve(tribeReserveStabilizer.address, feiTokensExchange);
+
+      await tribeReserveStabilizer.updateOracle();
+
+      await tribeReserveStabilizer.startOracleDelayCountdown();
+      await time.increase(await tribeReserveStabilizer.duration());
+
+      await collateralizationOracleWrapper.update();
+
+      const expectedAmountOut = await tribeReserveStabilizer.getAmountOut(feiTokensExchange);
+      await tribeReserveStabilizer.exchangeFei(feiTokensExchange);
+
+      const userFeiBalanceAfter = toBN(await fei.balanceOf(deployAddress));
+      const userTribeBalanceAfter = await tribe.balanceOf(deployAddress);
+
+      expect(userTribeBalanceAfter.sub(toBN(expectedAmountOut))).to.be.equal(userTribeBalanceBefore);
```

Comment:
I think this test should hand calculated expectedAmountOut; example bug that this wouldn't catch is if expectedAmountOut is zero and it doesn't mint anything.

(either that or unit test expected amount out, which should be unit tested anyway)

Reply:
its unit tested in ReserveStabilizer